lavalink/models.py

```diff
@@ -279,9 +279,11 @@ def add(self, requester: int, track: typing.Union[dict, AudioTrack], extra: dict
                 return
 
             if self.shuffle:
-                track = self.queue.pop(randrange(len(self.queue)))
+                track = self.queue.index(randrange(len(self.queue)))
+                self.queue.remove(track)
             else:
-                track = self.queue.pop(0)
+                track = self.queue.index(0)
+                self.queue.remove(track)
```

Comment:
Dequeue implements `popleft()`, it does the same thing as you two lines. But way faster.